core/src/main/resources/lib/hudson/executors.jelly

```diff
@@ -110,7 +110,11 @@ THE SOFTWARE.
                 </st:include>
                 <td class="pane" align="center" valign="middle">
                   <j:if test="${e.hasStopPermission()}">
-                    <l:stopButton href="${rootURL}/${c.url}${url}/stop" confirm="${%confirm(exe.fullDisplayName)}" alt="${%terminate this build}"  />
+                    <j:invokeStatic var="runId" className="java.net.URLEncoder" method="encode">
+                      <j:arg value="${exe.externalizableId}" />
+                      <j:arg value="UTF-8" />
+                    </j:invokeStatic>
+                    <l:stopButton href="${rootURL}/${c.url}${url}/stopBuild?runExtId=${runId}" confirm="${%confirm(exe.fullDisplayName)}" alt="${%terminate this build}"  />
```

Comment:
You can just use `${h.encode(exe.externalizableId)}` to simplify the URL encoding.

Reply:
`h.encode` is [Functions.encode](https://github.com/jenkinsci/jenkins/blob/jenkins-2.199/core/src/main/java/hudson/Functions.java#L725) which is [Util.encode](https://github.com/jenkinsci/jenkins/blob/jenkins-2.199/core/src/main/java/hudson/Util.java#L800), so it's not safe to encode a query parameter. In particular, `#` would not be encoded, and thus the build number would be stripped (treated as an anchor by the browser).

[Util.rawEncode](https://github.com/jenkinsci/jenkins/blob/jenkins-2.199/core/src/main/java/hudson/Util.java#L856) would be better (at least `#` would be encoded), but it's not available from `Functions`/`h`, and it would still break on some silly job names (like `a=a&b=b`, because `&` would not be encoded - but yeah, the UI discourages usage of such silly names, so it would still be acceptable I guess). I've seen it used in one jelly file ([FileParameterValue/value.jelly](https://github.com/jenkinsci/jenkins/blob/jenkins-2.199/core/src/main/resources/hudson/model/FileParameterValue/value.jelly#L32)), but it is via `j:invokeStatic` too, so not really simpler than calling `URLEncoder.encode`.

Reply:
Ah, good point. Perhaps you could add a `urlEncode` method to `Functions`? No big deal either way.

Reply:
> Perhaps you could add a `urlEncode` method to `Functions`?

Let's do that in an other PR: #4278
I will update this one if it gets merged.